stacks-core/src/address.rs

```diff
@@ -0,0 +1,253 @@
+use std::fmt;
+
+use crate::{
+    c32::{decode_address, encode_address},
+    crypto::{
+        hash::{Hash160, SHA256Hash},
+        PublicKey,
+    },
+    StacksError, StacksResult,
+};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum AddressHashMode {
+    SerializeP2PKH,
+    SerializeP2SH,
+    SerializeP2WPKH,
+    SerializeP2WSH,
+}
+
+#[derive(Debug, Clone)]
+pub struct StacksAddress {
+    version: u8,
```

Comment:
Can we have a strong type for the version instead of a u8?

Reply:
Yes, that's been especially painful now when working on StacksAddress tests.